torch/csrc/jit/serialization/export_bytecode.cpp

```diff
@@ -248,7 +252,54 @@ std::unique_ptr<mobile::Code> compileGraphToMobileCode(
   return mobile_code_ptr;
 }
 
-void checkSchema(const FunctionSchema& schema) {
+std::unique_ptr<mobile::Function> convertJitFunctionToMobileFunction(const GraphFunction *function, const CompilationOptions& options) {
+  std::shared_ptr<mobile::Code> mobileCode = compileGraphToMobileCode(function->name(), function->graph(), options);
+  const auto& schema = function->getSchema();
+  auto mobile_func = std::make_unique<mobile::Function>(
+        function->qualname(), mobileCode, schema);
+
+  return mobile_func;
+}
+
+IValue convertMobileFunctionToCodeTable(const mobile::Function *func, const CompilationOptions& compilation_options){
+  const std::shared_ptr<mobile::Code> code = func->get_code();
+  std::vector<IValue> instructions;
+  instructions.reserve(code->instructions_.size());
+  for (Instruction ins : code->instructions_) {
+    instructions.emplace_back(to_tuple({toString(ins.op), ins.X, ins.N}));
+  }
+
+  std::vector<IValue> operators;
+  operators.reserve(code->op_names_.size());
+  for (int i = 0; i < code->op_names_.size(); ++i) {
+    const auto& opname = code->op_names_[i];
+    const int size = code->operator_input_sizes_[i];
+    if (compilation_options.enable_default_value_for_unspecified_arg) {
+      operators.emplace_back(to_tuple({opname.name, opname.overload_name}));
+    } else {
+      operators.emplace_back(
+          to_tuple({opname.name, opname.overload_name, size}));
+    }
+  }
+
+  std::vector<IValue> types;
+  for (const TypePtr& t : code->types_) {
```

Comment:
Probably leave a comment here, explaining NameTuple is not supported in upgrader bytecode at the moment. Ideally it's better to throw an exception if we see a NamedTuple here.